### ISSUE TITLE
Validating Email

### ISSUE THREAD

**Issue by Bhattu-Anitha-07** [Anishkagupta04/RAPIDOC-HEALTHCARE-WEBSITE-#949]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Validating email, register the user if the entered email is valid.

### Use Case

Free from unauthorized login.

### Add ScreenShots

![image](https://github.com/user-attachments/assets/e2e02aa6-6387-4df5-8e2e-2e6ad0a6973b)

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

**Comment by Bhattu-Anitha-07** [comment]
@Anishkagupta04 @varshith257 
Please assign this issue to me,
will work on it. Also label the issue
Thank you.

**Comment by Bhattu-Anitha-07** [comment]
@varshith257 
I have a doubt, Can you please verify?

**Comment by varshith257** [comment]
Yes, go ahead

**Comment by Bhattu-Anitha-07** [comment]
I have created an account in firebase and used it for login authentication.
Can I use my account or is there any problem?

**Comment by varshith257** [comment]
I think firebase isn't fully supported

**Comment by varshith257** [comment]
Better use a library to validate email in client side like `email-validator`.

There could be multiple approaches for it but I recommend above.

For guide you can follow this article of method 3

https://piyush132000.medium.com/mastering-email-validation-in-javascript-multiple-approaches-ae718546160b

**Comment by Bhattu-Anitha-07** [comment]
But these kind of approach allow users with invalid email id to create an account, which is not an efficient way.
So that's the reason I have used this firebase.

**Comment by varshith257** [comment]
But we are not still yet to support server side

**Comment by Bhattu-Anitha-07** [comment]
No need of any server side support.
Firebase is a platform by google which provides backend services such as real-time databases, authentication etc.

**Comment by Bhattu-Anitha-07** [comment]
Project called Swap reads used firebase for authentication, So can I go with it.

**Comment by varshith257** [comment]
Ok, go with it for auth services

@Pruthaa17 We have a discussion of db support a long ago. We can go with firebase for now and see how it forms later we check with adding server side with PostgreeSQL

**Comment by Bhattu-Anitha-07** [comment]
Thankyou @varshith257, for the clarification.
Using my account for authentication, if possible you can create an account and provide configuration details.

**Comment by varshith257** [comment]
Create .env file for credentials and  provide demo video of its working with your credentials and  while submitting pr remove your credentials and also update readme with the same or maybe in CONTRIBUTING.md file showing how to to setup firebase account locally to use by other contributers

**Comment by Bhattu-Anitha-07** [comment]
Ok @varshith257

**Comment by Pruthaa17** [comment]
> Ok, go with it for auth services
> 
> @Pruthaa17 We have a discussion of db support a long ago. We can go with firebase for now and see how it reflects later we check with adding server side with PostgreeSQL

Sure , Right now do you have task considering database support that I can work with ??

**Comment by varshith257** [comment]
Yes, we start with adding db schemas to firebase and validate the inputs or store info provided by user

**Comment by Pruthaa17** [comment]
Okay

**Comment by Bhattu-Anitha-07** [comment]
@varshith257 
Sorry, I am unable resolve this issue.

**Comment by varshith257** [comment]
Anything obstructing the progress of this work?

**Comment by Bhattu-Anitha-07** [comment]
In Firebase, it is showing some cross-origin warnings which I am unable to understand.
